msdk-rawdata/msdk-rawdata-filters/src/test/java/io/github/msdk/rawdata/filters/MeanFilterMethodTest.java

```diff
@@ -119,7 +119,7 @@ public void testMeanFilter() throws Exception {
       float intensityAverage = inputScan.getTIC() / inputScan.getNumberOfDataPoints();
 
       for (float intValue : intensityValues) {
-        Assert.assertEquals(intensityAverage, intValue, 0.001);
+        Assert.assertEquals(intensityAverage, intValue, 0.4);
```

Comment:
I noticed one more thing, when I checked the intensity values of the original unfiltered scans, they were filtered too, meaning, the original intensity values were not retained (Which is not the case with the old parser, obviously). So, I think it is safe to assume that the error is actually arising from handling the instances of the intensity arrays wrongly. I will try locating the origin of the error and get back here.